### PR TITLE
Graphic link feature

### PR DESCRIPTION
**Pull request by guywillis**
Resolves https://github.com/adaptlearning/adapt_framework/issues/2832

* Allows the graphic to be a clickable link 
* Link can be an Adapt id reference (#/id/c-10), external url (https://www.adaptlearning.org/), or relative file (course/en/images/vanilla-swatch.jpg)

### CODE REVIEW

**Comment by oliverfoster** [comment]
@guywillis this is now a jsx component, so you may have to start again?

**Comment by oliverfoster** [comment]
@guywillis yeet

File: README.md [hunk 1]
```diff
@@ -36,39 +36,41 @@ The attributes listed below are used in *components.json* to configure **Graphic
...
 
->**\_url** (string): When the graphic is clicked this is the url it will follow. Can be an Adapt reference, `#/id/co-10`, external, `https://www.adaptlearning.org/`, or file `course/en/images/vanilla-swatch.jpg`. If using an Adapt reference `_target` must be set to `_self`.
+>**\_url** (string): When the graphic is selected this is the url it will follow. Can be an Adapt reference, `#/id/co-10`, external, `https://www.adaptlearning.org/`, or file `course/en/images/vanilla-swatch.jpg`. If using an Adapt reference `_target` must be set to `_self`.
 
->**\_target** (string): The url will open on target. Acceptable values are `_blank`, `_parent`, `_self` or `_top`.
+>**\_target** (string): The target attribute specifies where to open the link or linked document. Acceptable values are `_blank` (opens the linked document in a new window or tab), `_parent` (opens the linked document in the parent frame), `_top` (opens the linked document in the full body of the window) or `_self` (opens the linked document in the same frame as it was selected. If no value is set, the default is `_blank`.
```
Comment (oliverfoster) [inline README.md:56]: brackets aren't needed.

File: js/GraphicView.js [hunk 2]
```diff
@@ -2,13 +2,39 @@ import ComponentView from 'core/js/views/componentView';
...
       this.setupInviewCompletion('.graphic__widget');
     });
   }
 
+  onClick(event) {
+    if (event) event.preventDefault();
+
+    const item = this.model.get('_graphic');
+    const url = item._url;
+    const target = item._target || '_blank';
```
Comment (oliverfoster) [inline js/GraphicView.js:23]:
```suggestion
    const { _url: url, _target: target = '_blank' } = item;
```

File: js/GraphicView.js [hunk 3]
```diff
@@ -2,13 +2,39 @@ import ComponentView from 'core/js/views/componentView';
...
+        if (url.substr(0, 1) === '#') {
+          Backbone.history.navigate(url, { trigger: true });
+          return;
+        }
+        window.location.href = url;
+        break;
+      default:
+        window.open(url, target);
+    }
+  }
```
Comment (oliverfoster) [inline js/GraphicView.js:36]:
```suggestion
    const isNewWindow = (target !== '_self');
    if (isNewWindow) return window.open(url, target);
    const isRouterNavigation = (url.substr(0, 1) === '#');
    if (isRouterNavigation) return Backbone.history.navigate(url, { trigger: true });
    window.location.href = url;
  }
```

**Review by oliverfoster — CHANGES_REQUESTED** [review]

**Review by kirsty-hames — APPROVED** [review]

**Pull request merged**